Mathlib/CategoryTheory/Adjunction/Basic.lean

```diff
@@ -482,6 +482,15 @@ instance rightAdjointOfComp {E : Type u₃} [Category.{v₃} E] {F : C ⥤ D} {G
 
 end
 
+section mono_epi
+
+instance (adj : F ⊣ G) {X : C} {Y : D} (f : F.obj X ⟶ Y) [Mono f] [Faithful F] :
+    Mono (adj.homEquiv _ _ f) where
+  right_cancellation _ _ h := F.map_injective
+    (by simpa only [← cancel_mono f] using (adj.homEquiv _ _).injective (by simpa using h))
```

Comment:
This one can be generalized to [F.ReflectsMonomorphisms] as I did in my commit. Should we keep this?

Reply:
I think we should just keep the version with `[F.ReflectsMonomorphisms]`

Reply:
Note that it has to live in the file EpiMono though, where ReflectsMonomorphisms is defined.

Reply:
done